integrations/issue_test.go

```diff
@@ -87,7 +89,12 @@ func TestViewIssuesKeyword(t *testing.T) {
 	defer prepareTestEnv(t)()
 
 	repo := models.AssertExistsAndLoadBean(t, &models.Repository{ID: 1}).(*models.Repository)
-
+	issue := models.AssertExistsAndLoadBean(t, &models.Issue{
+		RepoID: repo.ID,
+		Index:  1,
+	}).(*models.Issue)
+	issues.UpdateIssueIndexer(issue)
+	time.Sleep(time.Second * 1)
```

Comment:
Mmm.... isn't this value too arbritrary? If it's intended to give the indexer some time to process the request, the value might be insufficient when the load is too heavy on the system. Perhaps we'll need some kind of flush mechanism in the future.

Reply:
So this is only a test but you're right.

Yeah we definitely need a flush for tests - or some other way of indicating that the job is done. That could be with a handler that writes to a test channel etc.

Reply:
However they interface doesn't really give us that option so hence the wait. 

The indexer in this case has just been started and so a second is not too unreasonable.

Reply:
If we want this to work properly we need to add a blevetest option - whereby you can add watchers to the various methods of the indexer with registerable callbacks à la https://github.com/go-gitea/gitea/blob/c6e1111934aa6350dff04882ca4fe02bba2d61d0/integrations/notification_helper_test.go

I think there is a problem with the way gitea is structured in general - it's not really possible to mock up things properly. It's the reason why our integration tests are fundamentally racy even if we manage to bodge a pass on the race test.